core/src/main/scala/kafka/server/ApiVersionManager.scala

```diff
@@ -70,35 +67,24 @@ class SimpleApiVersionManager(
 
 class DefaultApiVersionManager(
   val listenerType: ListenerType,
-  interBrokerProtocolVersion: MetadataVersion,
   forwardingManager: Option[ForwardingManager],
   features: BrokerFeatures,
-  featureCache: FinalizedFeatureCache
+  metadataCache: MetadataCache
 ) extends ApiVersionManager {
 
   override def apiVersionResponse(throttleTimeMs: Int): ApiVersionsResponse = {
     val supportedFeatures = features.supportedFeatures
-    val finalizedFeaturesOpt = featureCache.get
+    val finalizedFeatures = metadataCache.features()
     val controllerApiVersions = forwardingManager.flatMap(_.controllerApiVersions)
 
-    finalizedFeaturesOpt match {
-      case Some(finalizedFeatures) => ApiVersionsResponse.createApiVersionsResponse(
+    ApiVersionsResponse.createApiVersionsResponse(
         throttleTimeMs,
-        interBrokerProtocolVersion.highestSupportedRecordVersion,
+        metadataCache.metadataVersion().highestSupportedRecordVersion,
         supportedFeatures,
         finalizedFeatures.features.map(kv => (kv._1, kv._2.asInstanceOf[java.lang.Short])).asJava,
         finalizedFeatures.epoch,
         controllerApiVersions.orNull,
         listenerType)
-      case None => ApiVersionsResponse.createApiVersionsResponse(
-        throttleTimeMs,
-        interBrokerProtocolVersion.highestSupportedRecordVersion,
-        supportedFeatures,
-        Collections.emptyMap(),
-        ApiVersionsResponse.UNKNOWN_FINALIZED_FEATURES_EPOCH,
```

Comment:
This empty case is handled by the `features()` method on MetadataCache

Reply:
excellent. nice to not have this weird special case